cgen: error C2099: initializer is not a constant

### Describe the bug

Fixed arrays like bellow running on `./v -os windows -cc msvc ./la.v -o la.c` return a error
'const unicode_escape_chars_b = [`\\`, `u`]!'

this gen
```c
typedef rune Array_fixed_rune_2 [2];
Array_fixed_rune_2 _const_main__unicode_escape_chars_b = {'\\', 'u'}; // fixed array const
```

### Reproduction Steps

run `./v -os windows -cc msvc ./la.v -o la.c`
on  file `la.v` on windows
```v
const quote_byte = `"`

const escaped_chars = [(r'\b'), (r'\f'), (r'\n'), (r'\r'), (r'\t')]!

// Array_rune _const_main__unicode_escape_chars; // inited later
const unicode_escape_chars_a = [`\\`, `u`]

// typedef rune Array_fixed_rune_2 [2];
// Array_fixed_rune_2 _const_main__unicode_escape_chars = {'\\', 'u'}; // fixed array const
const unicode_escape_chars_b = [`\\`, `u`]!
```

### Expected Behavior

no error

### Current Behavior

`error C2099: initializer is not a constant`

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.4.3 85bf157

### Environment details (OS name and version, etc.)

Ubuntu
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -5362,7 +5362,8 @@ fn (mut g Gen) const_decl(node ast.ConstDecl) {
 		field_expr := field.expr
 		match field.expr {
 			ast.ArrayInit {
-				if field.expr.is_fixed && g.pref.build_mode != .build_module {
+				if field.expr.is_fixed && g.pref.build_mode != .build_module
+					&& (!g.is_cc_msvc || field.expr.elem_type != ast.string_type) {
 					styp := g.typ(field.expr.typ)
 					val := g.expr_string(field.expr)
 					g.global_const_defs[util.no_dots(field.name)] = GlobalConstDef{
@@ -5370,6 +5371,10 @@ fn (mut g Gen) const_decl(node ast.ConstDecl) {
 						def: '${styp} ${const_name} = ${val}; // fixed array const'
 						dep_names: g.table.dependent_names_in_expr(field_expr)
 					}
+				} else if field.expr.is_fixed && g.is_cc_msvc
+					&& field.expr.elem_type == ast.string_type {
+					g.const_decl_init_later_msvc_string_fixed_array(field.mod, name, field.expr,
+						field.typ)
 				} else {
 					g.const_decl_init_later(field.mod, name, field.expr, field.typ, false,
 						false)
@@ -5656,6 +5661,30 @@ fn (mut g Gen) const_decl_init_later(mod string, name string, expr ast.Expr, typ
 			g.cleanup.writeln('\tmap_free(&${cname});')
 		} else if styp == 'IError' {
 			g.cleanup.writeln('\tIError_free(&${cname});')
+		}
+	}
+}
+
+fn (mut g Gen) const_decl_init_later_msvc_string_fixed_array(mod string, name string, expr ast.ArrayInit, typ ast.Type) {
+	mut styp := g.typ(typ)
+	cname := g.c_const_name(name)
+	mut init := strings.new_builder(100)
+	for i, elem_expr in expr.exprs {
+		init.writeln(g.expr_string_surround('\t${cname}[${i}] = ', elem_expr, ';'))
+	}
+	mut def := '${styp} ${cname}'
+	g.global_const_defs[util.no_dots(name)] = GlobalConstDef{
+		mod: mod
+		def: '${def}; // inited later'
+		init: init.str().trim_right('\n')
+		dep_names: g.table.dependent_names_in_expr(expr)
+	}
+	if g.is_autofree {
+		sym := g.table.sym(typ)
+		if sym.has_method_with_generic_parent('free') {
+			g.cleanup.writeln('\t${styp}_free(&${cname});')
+		} else {
+			g.cleanup.writeln('\tarray_free(&${cname});')
 		}
 	}
 }

--- a/vlib/v/tests/const_fixed_array_of_string_value_msvc_test.v
+++ b/vlib/v/tests/const_fixed_array_of_string_value_msvc_test.v
@@ -1,0 +1,9 @@
+// for issue 20287
+// msvc does not support the "_SLIT" macro for fixed array of string values immediate initialization:
+// Array_fixed_string_5 _const_main__escaped_chars = {(_SLIT("\\b")), (_SLIT("\\f")), (_SLIT("\\n")), (_SLIT("\\r")), (_SLIT("\\t"))};
+// error C2099: initializer is not a constant
+const escaped_chars = [(r'\b'), (r'\f'), (r'\n'), (r'\r'), (r'\t')]!
+
+fn test_main() {
+	assert true
+}

